packages/scenario/src/client/init.ts

```diff
@@ -0,0 +1,104 @@
+import type { XFormsElement } from '@odk-web-forms/common/test/fixtures/xform-dsl/XFormsElement.ts';
+import {
+	initializeForm,
+	type EngineConfig,
+	type FormResource,
+	type RootNode,
+} from '@odk-web-forms/xforms-engine';
+import type { Owner } from 'solid-js';
+import { createRoot, getOwner, runWithOwner } from 'solid-js';
+import { FormDefinitionResource } from '../jr/resource/FormDefinitionResource.ts';
+
+/**
+ * @todo It's anticipated that this will be expanded to support the various ways
+ * that JavaRosa's tests load form resources. This initially supports the (also
+ * ported) in-source DSL. We'll likely also need to support loading fixtures by
+ * file name (and various ambiguities that come with that, which we may want to
+ * disambiguate as we port).
+ */
+export type TestFormResource = FormDefinitionResource | XFormsElement;
+
+const isPathResource = (resource: TestFormResource): resource is FormDefinitionResource => {
+	return resource instanceof FormDefinitionResource;
+};
+
+const isXFormsElement = (resource: TestFormResource): resource is XFormsElement => {
+	return typeof (resource as XFormsElement).asXml === 'function';
+};
+
+export const getFormResource = async (
+	testFormResource: TestFormResource
+	// It's also anticipated that we will need to perform async IO as we expand
+	// the `TestFormResource` type.
+	// eslint-disable-next-line @typescript-eslint/require-await
+): Promise<FormResource> => {
+	if (isPathResource(testFormResource)) {
+		return testFormResource.textContents;
+	}
+
+	if (isXFormsElement(testFormResource)) {
+		return testFormResource.asXml();
+	}
+
+	throw new Error('Unknown test form resource');
+};
+
+/**
+ * @todo Currently we stub resource fetching. We can address this as needed
+ * while we port existing tests and/or add new ones which require it.
+ */
+const fetchResourceStub: typeof fetch = () => {
+	throw new Error('TODO: resource fetching not implemented');
+};
+
+/**
+ * Satisfies the xforms-engine client `stateFactory` option. Currently this is
+ * intentionally **not** reactive, as the current scenario tests (as
+ * ported/derived from JavaRosa's test suite) do not explicitly exercise any
+ * reactive aspects of the client interface.
+ *
+ * @todo It **is possible** to use Solid's `createMutable`, which would enable
+ * expansion of the JavaRosa test suite to _also_ test reactivity. In local
+ * testing during the migration to the new client interface, no additional
+ * changes were necessary to make that change. For now this non-reactive factory
+ * is supplied as a validation that reactivity is in fact optional (despite the
+ * fact that we're kind of observing the internal reactivity with a couple of
+ * memos below).
+ */
```

Comment:
I'm removing this parenthetical. It referenced earlier init logic, presumably around earlier/more naive mapping of engine nodes to the JR/Scenario concept of "question"/"event".

However! I was curious about any remaining memos that this might reference. It turns out there's only one `createMemo` call left from that earlier pass, which is [here](https://github.com/getodk/web-forms/blob/d40bd88ed8959bbe7fae5d28efc840c16ca50a72/packages/scenario/src/jr/Scenario.ts#L163-L173). My instinct was to unwrap this as a plain thunk. Doing that caused three tests to pass which are currently known failures/marked `it.fails`. Each of those known failures includes a note that they fail due to the engine producing an [`InconsistentChildrenStateError`](https://github.com/getodk/web-forms/blob/d40bd88ed8959bbe7fae5d28efc840c16ca50a72/packages/xforms-engine/src/lib/reactivity/materializeCurrentStateChildren.ts#L72-L75).

This is actually consistent with what I had previously observed/hypothesized when I saw the error in `ui-solid`. If I recall, in that case the error was only produced alongside use of Solid's [control flow helpers](https://github.com/solidjs/solid/blob/a72d393a07b22f9b7496e5eb93712188ccce0d28/packages/solid/src/render/flow.ts), which also have several calls into `createMemo`.

So two bits of good news here:

1. Whatever bug exists there probably only affects clients using Solid's `createMutable` as the reactive client state factory, and then appears to be triggered by some interaction between `createMemo` outside of the engine's own tracking scopes. As such, I'm also updating the notes on each of those tests to reflect this discovery.
2. This accounts for all of the cases of `InconsistentChildrenStateError` mentioned in the porting notes, so it seems pretty likely that there's only one bug and not some hydra of unknown behavior.

It would be tempting to unwrap the `createMemo` call in Scenario and let those tests pass now. But I'd like to continue tracking the failure case so we can potentially have greater confidence in the opaque reactivity API, in case there's ever an appetite for a full fledged Solid client (and as a general measure to keep ourselves honest about the generality of that aspect of the interface).